src/test/java/edu/unc/lib/boxc/migration/cdm/services/RedirectMappingIndexServiceTest.java

```diff
@@ -0,0 +1,160 @@
+import org.apache.commons.csv.CSVRecord;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.Reader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.Statement;
+import java.util.ArrayList;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * @author snluong
+ */
+public class RedirectMappingIndexServiceTest {
+    private static final String PROJECT_NAME = "proj";
+    private static final String USERNAME = "migr_user";
+    private static final String DEST_UUID = "7a33f5e6-f0ca-461c-8df0-c76c62198b17";
+
+    @Rule
+    public final TemporaryFolder tmpFolder = new TemporaryFolder();
+
+    private MigrationProject project;
+    private RedirectMappingIndexService indexService;
+    private SipService sipsService;
+    private SipServiceHelper testHelper;
+
+    @Before
+    public void setup() throws Exception {
+        tmpFolder.create();
+        project = MigrationProjectFactory.createMigrationProject(
+                tmpFolder.getRoot().toPath(), PROJECT_NAME, null, "user");
+        testHelper = new SipServiceHelper(project, tmpFolder.newFolder().toPath());
+        sipsService = testHelper.createSipsService();
+
+        indexService = new RedirectMappingIndexService(project);
+        Connection conn = indexService.openDbConnection();
+        Statement statement = conn.createStatement();
+        statement.execute("drop table if exists redirect_mappings");
+        statement.execute("CREATE TABLE redirect_mappings (" +
+                "id INTEGER PRIMARY KEY AUTOINCREMENT, " +
+                "cdm_collection_id varchar(64) NOT NULL, " +
+                "cdm_object_id varchar(64) NOT NULL, " +
+                "boxc_work_id varchar(64) NOT NULL, " +
+                "boxc_file_id varchar(64) DEFAULT NULL, " +
+                "created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP, " +
+                "UNIQUE (cdm_collection_id, cdm_object_id)" +
+                ")");
+        CdmIndexService.closeDbConnection(conn);
+
+        testHelper.indexExportData("export_1.xml");
+        testHelper.generateDefaultDestinationsMapping(DEST_UUID, null);
+        testHelper.populateDescriptions("gilmer_mods1.xml");
+        testHelper.populateSourceFiles("276_182_E.tif", "276_183B_E.tif", "276_203_E.tif");
+
+        sipsService.generateSips(makeOptions());
+    }
+
+    @Test
+    public void indexingDoesNotHappenIfCollectionIsNotSubmitted() {
+        try {
+            indexService.indexMapping();
+        } catch (InvalidProjectStateException e) {
+            assertTrue("Unexpected message: " + e.getMessage(),
+                    e.getMessage().contains("Must submit the collection prior to indexing"));
+        }
+    }
+
+    @Test
+    public void redirectMappingTableIsPopulatedWithRightNumberOfRows() throws Exception {
```

Comment:
You could merge this test with the next one, they are both verifying that the same set of redirects mapping made it into the database, using two different metrics to verify it. I generally try to base tests around the state/condition that is being tested rather than the way the outcome is validated. If the test is getting super long then it can make sense to split it up or use private helper methods to improve readability, but i think in this case it'd be okay to merge.

Reply:
ok, got it